src/net/server/handlers/mul.rs

```diff
@@ -192,16 +197,243 @@ mod tests {
 
         // try a request 10 times
         for _ in 0..10 {
-            send_req(
-                &mut rx,
-                port,
-                query_id,
-                step.clone(),
-                target_helper,
-                offset,
-                body,
-            )
-            .await;
+            let resp = send_req(port, query_id, step.clone(), target_helper, offset, body).await;
+
+            let status = resp.status();
+            let resp_body = body::to_bytes(resp.into_body()).await.unwrap();
+            let resp_body_str = String::from_utf8_lossy(&resp_body);
+
+            // response comparison
+            let channel_id = ChannelId {
+                identity: target_helper,
+                step: step.clone(),
+            };
+            let env = [0; DATA_SIZE as usize].to_vec().into_boxed_slice();
+            let envs = (0..DATA_LEN)
+                .map(|i| MessageEnvelope {
+                    record_id: i.into(),
+                    payload: env.clone(),
+                })
+                .collect::<Vec<_>>();
+
+            assert_eq!(status, StatusCode::OK, "{}", resp_body_str);
+            let messages = rx.try_recv().expect("should have already received value");
+            assert_eq!(messages, (channel_id, envs));
+        }
+    }
+
+    #[allow(clippy::mutable_key_type)] // `HeaderName` is known good key
+    fn build_malformed_req<Q: Into<String>, S: Into<String>, I: Into<String>, B: Into<Body>>(
```

Comment:
this function has the ability to construct any request, not just malformed, so maybe just `post_request`?

Reply:
you are right, but I reworked this test, and this fn was removed